tools/generate-user-messages-screenshot

```diff
@@ -0,0 +1,197 @@
+            DEFAULT_USER, sub, stream, "color", color, acting_user=DEFAULT_USER
+        )
+
+
+def send_stream_messages(
+    stream_name: str, topic: str, staged_messages_data: List[MessageThread]
+) -> List[int]:
+    staged_messages = [dict(staged_message) for staged_message in staged_messages_data]
+    stream = ensure_stream(realm, stream_name, acting_user=DEFAULT_USER)
+    subscribers_query = get_active_subscriptions_for_stream_id(
+        stream.id, include_deactivated_users=False
+    ).values_list("user_profile", flat=True)
+
+    subscribers: Dict[str, UserProfile] = {}
+    for subscriber_id in subscribers_query:
+        subscriber = UserProfile.objects.get(realm=realm, id=subscriber_id)
+        subscribers[subscriber.full_name] = subscriber
+
+    messages = [
+        internal_prep_stream_message(
+            subscribers[message["sender"]],
+            stream,
+            topic,
+            message["content"],
+        )
+        for message in staged_messages
+    ]
+
+    message_ids = [
+        sent_message_result.message_id for sent_message_result in do_send_messages(messages)
+    ]
+    global message_thread_ids
+    message_thread_ids += message_ids
+
+    for message, message_id in zip(staged_messages, message_ids):
+        if message.get("reactions") is not None:
+            reactions = message["reactions"]
+            for reaction, user_names in reactions.items():
+                users = [subscribers[user_name] for user_name in user_names]
+                add_message_reactions(message_id, reaction, users)
+
+    return message_ids
+
+
+def add_message_reactions(message_id: int, emoji: str, users: List[UserProfile]) -> None:
+    preview_message = access_message(user_profile=DEFAULT_USER, message_id=message_id)
+    emoji_data = get_emoji_data(realm.id, emoji)
+    for user in users:
+        do_add_reaction(
+            user, preview_message, emoji, emoji_data.emoji_code, emoji_data.reaction_type
+        )
+
+
+def capture_streams_narrow_screenshot(
+    image_path: str, stream_name: str, topic: str, unread_msg_id: int
+) -> None:
+    stream = ensure_stream(realm, stream_name, acting_user=DEFAULT_USER)
+
+    narrow_uri = topic_narrow_url(realm=realm, stream=stream, topic_name=topic)
+    narrow = f"{stream_name}/{topic}"
+    screenshot_script = os.path.join(TOOLS_DIR, "thread-screenshot.js")
+    subprocess.check_call(
+        ["node", screenshot_script, narrow_uri, narrow, str(unread_msg_id), image_path, realm.url]
+    )
+
+
+parser = argparse.ArgumentParser()
+group = parser.add_mutually_exclusive_group(required=True)
+
+group.add_argument(
+    "--thread",
+    nargs="+",
+    type=str,
+    help="Path of the file where the thread for screenshot is present",
+)
+fixture_group = parser.add_argument_group("thread")
+
+options = parser.parse_args()
+prepare_puppeteer_run()
+
+realm = get_realm("zulip")
+with open(options.thread[0]) as f:
+    threads = json.load(f)
```

Comment:
What does the configuration file look like? I'm not sure JSON is likely to be the most convenient format for drafting these; could imagine just using a set of Python dataclasses instead, each one declared in an individual file in a `tools/screenshots/` directory ... which probably we should create and move the existing tool into as a prep step for this PR...

Reply:
Oh, I see, it's the next commit; yeah let's at least do the directory structure change proposed in my comment to give this cluster of tools its own directory to avoid cluttering the `tools/` root with at least one file per screenshot, and then perhaps @alya can review and consider whether she's prefer a different format for composing these.

Reply:
updated!